spa/src/components/pageEditor/elementEditors/amount/AmountEditor.tsx

```diff
@@ -9,10 +9,38 @@ import { ElementDetailEditorProps } from 'components/pageEditor/editInterface/El
 export type AmountEditorProps = ElementDetailEditorProps<AmountElement['content']>;
 
 export function AmountEditor({ contributionIntervals, elementContent, onChangeElementContent }: AmountEditorProps) {
-  function handleAddAmount(interval: ContributionInterval, value: number) {
-    // Keep the list of amounts sorted numerically.
+  useEffect(() => {
+    // Migrate legacy data to the new format on first render.
+    if (elementContent.allowOther) {
+      onChangeElementContent({
+        ...elementContent,
+        options: {
+          ...Object.entries(elementContent.options).reduce(
+            (acc, [interval, options]) => ({
+              ...acc,
+              [interval]: options.includes('other') ? options : [...options, 'other']
+            }),
+            {}
+          )
+        },
```

Comment:
```suggestion
        options: Object.entries(elementContent.options).reduce(
            (acc, [interval, options]) => ({
              ...acc,
              [interval]: options.includes('other') ? options : [...options, 'other']
            }),
            {}
          ),
```

Don't need the outermost spread here.